clang/lib/AST/ExprConstant.cpp

```diff
@@ -2778,7 +2778,9 @@ static bool CheckedIntArithmetic(EvalInfo &Info, const Expr *E,
     if (Info.checkingForUndefinedBehavior())
       Info.Ctx.getDiagnostics().Report(E->getExprLoc(),
                                        diag::warn_integer_constant_overflow)
-          << toString(Result, 10) << E->getType() << E->getSourceRange();
+          << toString(Result, 10, Result.isSigned(), /*formatAsCLiteral=*/false,
+                      /*upperCase=*/true, /*insertSeparators=*/true)
```

Comment:
```suggestion
                      /*UpperCase=*/true, /*InsertSeparators=*/true)
```
Same changes needed elsewhere, but this should keep clang-tidy happy for folks using it.